base/test.jl

```diff
@@ -492,11 +494,25 @@ function finish(ts::DefaultTestSet)
     end
     println()
     # Recursively print a summary at every level
-    print_counts(ts, 0, align, pass_width, fail_width, error_width, broken_width, total_width)
+    print_counts(ts, depth_pad, align, pass_width, fail_width, error_width, broken_width, total_width)
+end
+
+# Called at the end of a @testset, behaviour depends on whether
+# this is a child of another testset, or the "root" testset
+function finish(ts::DefaultTestSet)
+    # If we are a nested test set, do not print a full summary
+    # now - let the parent test set do the printing
+    if get_testset_depth() != 0
+        # Attach this test set to the parent test set
+        parent_ts = get_testset()
+        record(parent_ts, ts)
+        return
+    end
     # Finally throw an error as we are the outermost test set
-    if total != total_pass
+    #print_test_results(ts)
```

Comment:
Ah I see. I was thinking that the solution for the Base testing setting would be making a different kind of test set, or as you say, maybe `DefaultTestSet` can have an option.